aws/resource_aws_lb.go

```diff
@@ -389,7 +389,16 @@ func resourceAwsLbUpdate(d *schema.ResourceData, meta interface{}) error {
 
 	}
 
-	if d.HasChange("subnets") {
+	if d.HasChange("subnets") && !d.IsNewResource() {
+		// subnets are assigned at Create; the 'change' here is an empty map for old
+		// and current subnets for new, so this change is redundant when the
+		// resource is just created.
+		if d.Get("load_balancer_type").(string) == "network" {
+			// Load Balancers of type 'network' cannot update their subnets at this
+			// time.
+			return fmt.Errorf("Unable to update subnets for loadbalancer %s, updating subnets is not supported for type network. Please use the 'taint' command to force a recreation of this resource", d.Id())
```

Comment:
Nitpick (read as _I'm happy to land this as is_): Could we possibly use the new `CustomizeDiff` to mark `subnets` as `ForceNew` in this case? https://github.com/hashicorp/terraform/pull/14887/files#diff-456d1e45f8a8d1585e8228597b9a2176R2845 instead of instructing the user to taint?

Reply:
TIL about `CustomizeDiff`! I've opened https://github.com/terraform-providers/terraform-provider-aws/pull/2310 as a replacement